src/FNA3D_Driver_Vulkan.c

```diff
@@ -6282,7 +6289,97 @@ void VULKAN_GetTextureData2D(
 	void* data,
 	int32_t dataLength
 ) {
-	/* TODO */
+	FNAVulkanRenderer *renderer = (FNAVulkanRenderer*) driverData;
+	VulkanTexture *vulkanTexture = (VulkanTexture*) texture;
+	ImageMemoryBarrierCreateInfo imageBarrierCreateInfo;
+	VulkanBuffer *stagingBuffer = vulkanTexture->stagingBuffer;
+	void *stagingData;
+	VkBufferImageCopy imageCopy;
+	int32_t row;
+	uint8_t *dataPtr = (uint8_t*) data;
+	int32_t formatSize = Texture_GetFormatSize(format);
+
```

Comment:
we probably want to explicitly end the render pass here, pipeline barrier might do it implicitly but if the image happens to already be in the correct format it won't

Reply:
actually i think we need a stall here, if the data is accessed right after this function the operation will not have executed until SwapBuffers is called

Reply:
oops i mean the stall should be at the bottom of this function, after vkCmdCopy is submitted